src/pool/cli.c

```diff
@@ -696,12 +736,8 @@ pool_rsvc_client_complete_rpc(struct dc_pool *pool, const crt_endpoint_t *ep,
 				      &out->po_hint);
 	D_MUTEX_UNLOCK(&pool->dp_client_lock);
 	if (rc == RSVC_CLIENT_RECHOOSE ||
-	    (rc == RSVC_CLIENT_PROCEED && daos_rpc_retryable_rc(out->po_rc))) {
-		rc = tse_task_reinit(task);
-		if (rc != 0)
-			return rc;
+	    (rc == RSVC_CLIENT_PROCEED && daos_rpc_retryable_rc(out->po_rc)))
```

Comment:
only the most minor note that is probably not worth changing - the comments preceding the function (here, and in container code function) indicate that if RSVC_CLIENT_RECHOOSE is returned, the task is reinited. Now with this change, I guess it means the task will be reinited by the caller.

Reply:
My bad. Will definitely fix the comment, probably in a follow-on doc-only PR to save time/CI resource.

Reply:
https://github.com/daos-stack/daos/pull/16174